document/field_dense_vector.go

```diff
@@ -0,0 +1,126 @@
+//  Copyright (c) 2022 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build densevector
+// +build densevector
+
+package document
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/blevesearch/bleve/v2/size"
+	index "github.com/blevesearch/bleve_index_api"
+)
+
+var reflectStaticSizeDenseVectorField int
+
+func init() {
+	var f DenseVectorField
+	reflectStaticSizeDenseVectorField = int(reflect.TypeOf(f).Size())
+}
+
+const DefaultDenseVectorIndexingOptions = index.IndexField
+
+type DenseVectorField struct {
+	name              string
+	dims              int
+	similarity        string
+	options           index.FieldIndexingOptions
+	value             []float32
+	numPlainTextBytes uint64
+}
+
+func (n *DenseVectorField) Size() int {
+	return reflectStaticSizeDenseVectorField + size.SizeOfPtr +
+		len(n.name) +
+		int(numBytesSingleFloats(n.value))
+}
+
+func (n *DenseVectorField) Name() string {
+	return n.name
+}
+
+func (n *DenseVectorField) ArrayPositions() []uint64 {
+	return nil
+}
+
+func (n *DenseVectorField) Options() index.FieldIndexingOptions {
+	return n.options
+}
+
+func (n *DenseVectorField) NumPlainTextBytes() uint64 {
+	return n.numPlainTextBytes
+}
+
+func (n *DenseVectorField) AnalyzedLength() int {
+	// dense vectors aren't analyzed
+	return 0
+}
+
+func (n *DenseVectorField) EncodedFieldType() byte {
+	return 'v'
+}
+
+func (n *DenseVectorField) AnalyzedTokenFrequencies() index.TokenFrequencies {
+	// dense vectors aren't analyzed
+	return nil
+}
+
+func (n *DenseVectorField) Analyze() {
+	// dense vectors aren't analyzed
```

Comment:
Since we are planning to add a separate field for UDA driven vector generation ( for the benefit of easy user interface and not cluttering the vector type),  we don't need a comment corresponding to UDA here in this PR.